boards/arm/stm32mp157c_dk2/doc/stm32mp157_dk2.rst

```diff
@@ -0,0 +1,302 @@
+  - Internal or external ADC/DAC reference VREF+
+
+- Graphics:
+
+  - 3D GPU: Vivante® - OpenGL® ES 2.0
+
+    - Up to 26 Mtriangle/s, 133 Mpixel/s
+
+  - LCD-TFT controller, up to 24-bit // RGB888
+
+    - up to WXGA (1366 × 768) @60 fps
+    - Two layers with programmable colour LUT
+
+  - MIPI® DSI 2 data lanes up to 1 GHz each
+
+- Timers:
+
+  - 2 × 32-bit timers with up to 4 IC/OC/PWM or pulse counter and quadrature (incremental) encoder input
+  - 2 × 16-bit advanced motor control timers
+  - 10 × 16-bit general-purpose timers (including 2 basic timers without PWM)
+  - 5 × 16-bit low-power timers
+  - RTC with sub-second accuracy and hardware calendar
+  - 2 × 4 Cortex®-A7 system timers (secure, non-secure, virtual, hypervisor)
+  - 1 × SysTick Cortex®-M4 timer
+
+- Hardware acceleration:
+
+  - AES 128, 192, 256, TDES
+  - HASH (MD5, SHA-1, SHA224, SHA256), HMAC
+  - 2 × true random number generator (3 oscillators each)
+  - 2 × CRC calculation unit
+
+- Debug mode:
+
+  - Arm® CoreSight™ trace and debug: SWD and JTAG interfaces
+  - 8-Kbyte embedded trace buffer
+  - 3072-bit fuses including 96-bit unique ID, up to 1184-bit available for user
+
+More information about STM32P157C can be found here:
+
+       - `STM32MP157C on www.st.com`_
+       - `STM32MP157C reference manual`_
+
+Supported Features
+==================
+
+The Zephyr stm32mp157c_dk2 board configuration supports the following hardware features:
+
++-----------+------------+-------------------------------------+
+| Interface | Controller | Driver/Component                    |
++===========+============+=====================================+
+| NVIC      | on-chip    | nested vector interrupt controller  |
++-----------+------------+-------------------------------------+
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
+
+The default configuration can be found in the defconfig file:
+
+	``boards/arm/stm32mp157c_dk2/stm32mp157c_dk2_defconfig``
+
+
+Connections and IOs
+===================
+
+STM32MP157C-DK2 Discovery Board schematic is available here: `STM32MP157C Discovery board schematics`_.
+
+
+System Clock
+------------
+
+The Cortex®-M4 Core is configured to run at a 209 MHz clock speed.
+
+Serial Port
+-----------
+
+The STM32MP157C-DK2 Discovery board has 8 U(S)ARTs.
+The Zephyr console output is assigned by default to the ram console to be dumped
+by the Linux Remoteproc Framework on Cortex®-A7 core. The UART 4 can enabled as
+Cortex®-M4 console.
+
+Programming and Debugging
+*************************
+The STM32MP157C doesn't have QSPI flash for the Cortex®-M4  and it needs to be
+started by the Cortex®-A7 core. The Cortex®-A7 core is responsible to load the
+Cortex®-M4 binary application into the RAM, get the Cortex®-M4 out of reset.
+The Cortex®-A7 can perform these steps at bootloader level or after the Linux
```

Comment:
Linux?  I suspect this should say ``Zephyr``, right?

Reply:
No it is Linux. The Cortex-M4 firmware is not independent. It has to be loaded by the Cortex-A7. 
In the ST distribution an Uboot and a Linux is running on Cortex-A7. So the Zephyr firmware is either loaded by Uboot or by the Linux.